### PR TITLE
feat(zfspv) add go lint target

### PR DESCRIPTION
**Pull request by vaniisgh**
**Why is this PR required? What issue does it fix?**:
adds a target for lint as needed by #165

**Questions**
should the golint output be formatted before printing somehow ... ?

also i need to fix the spelling for recommendations, will do with the change-log or any additional changes :) 

**Checklist:**
- [x] Fixes #<issue number>
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

**Comment by pawanpraka1** [comment]
@vaniisgh this looks good. are there any golint error you are getting? if yes, would you like to fix them in the coming PR? We will add `make golint` command in the travis yaml and will make it fail if there are any linting error. But before doing that, we have to fix all the linting errore we have now.

File: Makefile [hunk 1]
```diff
@@ -25,6 +25,7 @@ VETARGS?=-asmdecl -atomic -bool -buildtags -copylocks -methods \
 # targets or for development purposes
 EXTERNAL_TOOLS=\
 	golang.org/x/tools/cmd/cover \
+	github.com/golang/lint/golint\
```
Comment (pawanpraka1) [inline Makefile:28]: formatting : add a space at the end.

File: Makefile [hunk 2]
```diff
@@ -238,3 +239,10 @@ ci:
 deploy-images:
 	@DIMAGE="${IMAGE_ORG}/zfs-driver" ./buildscripts/push
 
+.PHONY: golint
+golint:
+	@echo "--> Running golint"
+	@echo "Consider these linter reccomendations"
```
Comment (pawanpraka1) [inline Makefile:245]: typo: recommendation

**Comment by vaniisgh** [comment]
so, I'll fix the other recommendations in this PR and the go-lint errors in a new one? :) right?

**Comment by pawanpraka1** [comment]
fix the recommendation given in this PR only. And for fixing the go-lint and adding it to the travis can be taken in a new PR.

**Comment by pawanpraka1** [comment]
Also, add the changelog file.

**Comment by vaniisgh** [comment]
I'll fix the linting errors & add to Travis in a new PR, should I open an issue for the same ... I'll take care of it in a day or two though :)

**Comment by pawanpraka1** [comment]
@vaniisgh yes, you can open an issue for that.

**Pull request merged**